ingester-bulk-protocol/src/main/java/org/apache/arrow/flight/BulkFlightClient.java

```diff
@@ -338,7 +355,10 @@ protected void waitUntilStreamReady() {
                     // If the stream is not ready, wait for a short time to avoid busy waiting
                     // This helps reduce CPU usage while still being responsive
                     try {
-                        this.onStreamReadyHandler.await(10, TimeUnit.MILLISECONDS);
+                        if (this.onStreamReadyHandler.await(10, TimeUnit.MILLISECONDS)) {
+                            // Allow some in-flight requests to be sent
+                            break;
```

Comment:
Why break? Stream may be not ready, sending requests could cause errors.

Reply:
We allow certain in-flight requests to be sent into the stream even if it is not ready. The request that obtained the permit can break the loop.